lib/sprockets/redirect.rb

```diff
@@ -74,14 +76,36 @@ def digest_path
 
     # Sends a redirect header back to browser
     def redirect_to_digest_version(env)
-      url = URI(@asset_host || @request.url)
-      url.path = "#{@prefix}/#{digest_path}"
+      url = File.join(compute_asset_host, @prefix, digest_path)
```

Comment:
`File.join` is not appropriate for URI, as this will join the URL with `\` if for some reason this is being run on Windows. Would you mind changing this to [`URI.join`](http://ruby-doc.org/stdlib-2.0.0/libdoc/uri/rdoc/URI.html#method-c-join)?